core/src/types/buffer.rs

```diff
@@ -149,6 +149,12 @@ impl Default for Buffer {
     }
 }
 
+impl AsRef<[u8]> for Buffer {
```

Comment:
Buffer could have multiple bytes. It's better not to implement this to avoid Buffer been used wrongly.

Reply:
Then we call `.chunk` everywhere?

Reply:
> Then we call `.chunk` everywhere?

Yes, if the consumer accepts `bytes::Buf`. In most cases, such as in this PR, we should collect them into a single `Bytes` to allow writing them in one request (like redis `set` call). 

We can optimize this by introducing `multi_write` for the KV adapter, similar to what we do with `oio::Write`.

Reply:
I suggest you push a new commit onto this branch. I can't get it quite clear.

And we may wait a bit for #4497  to get the `collect` update.